### DISCUSSION TITLE
Why do I have a weird table Name after migrating my prisma schema?

### DISCUSSION THREAD

**Discussion by SelfDevTV** [prisma/prisma#4927] [Q&A] [answered]
Hiho guys,

I started an issue a few days ago because my performance of my prisma nested queries are very slow #4884 
So I was trying out some thing to find errors. So one thing I tried is installing pgAdmin and try to run some queries and look how they perform. 
While trying this I found a weird table that I haven't anywhere in my prisma schema.
The weird table is this:

 ## _SubredditToUser

This is how it looks:

![image](https://user-images.githubusercontent.com/48333000/104072114-cc8ee700-520a-11eb-815b-25a25d6d1a7b.png)

### Where is this comming from and what does it do? On my query that is slow it also queries this (with `prisma-client` ==> Logging turned on):

>SELECT "public"."_SubredditToUser"."A", "public"."_SubredditToUser"."B" FROM "public"."_SubredditToUser" WHERE >"public"."_SubredditToUser"."A" IN ($1)

Is this some `prisma` internal stuff? But why exactly this name and where can I find it in the docs?

In the latest local migration file I found this:

```sql
-- CreateTable
CREATE TABLE "_SubredditToUser" (
    "A" INTEGER NOT NULL,
    "B" INTEGER NOT NULL
);
```

Really don't know where this comes from. 

Here is my schema file:

```prisma
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider = "prisma-client-js"
}

model Account {
  id                 Int       @id @default(autoincrement())
  compoundId         String    @unique @map(name: "compound_id")
  userId             Int       @map(name: "user_id")
  providerType       String    @map(name: "provider_type")
  providerId         String    @map(name: "provider_id")
  providerAccountId  String    @map(name: "provider_account_id")
  refreshToken       String?   @map(name: "refresh_token")
  accessToken        String?   @map(name: "access_token")
  accessTokenExpires DateTime? @map(name: "access_token_expires")
  createdAt          DateTime  @default(now()) @map(name: "created_at")
  updatedAt          DateTime  @default(now()) @map(name: "updated_at")


  @@index([providerAccountId], name: "providerAccountId")
  @@index([providerId], name: "providerId")
  @@index([userId], name: "userId")
  @@map(name: "accounts")
}

model Session {
  id           Int      @id @default(autoincrement())
  userId       Int      @map(name: "user_id")
  expires      DateTime
  sessionToken String   @unique @map(name: "session_token")
  accessToken  String   @unique @map(name: "access_token")
  createdAt    DateTime @default(now()) @map(name: "created_at")
  updatedAt    DateTime @default(now()) @map(name: "updated_at")

  @@map(name: "sessions")
}

model User {
  id            Int         @id @default(autoincrement())
  name          String?
  email         String?     @unique
  emailVerified DateTime?   @map(name: "email_verified")
  image         String?
  createdAt     DateTime    @default(now()) @map(name: "created_at")
  updatedAt     DateTime    @default(now()) @map(name: "updated_at")
  posts         Post[]
  comments      Comment[]
  votes         Vote[]
  joinedSubs    Subreddit[]

  @@map(name: "users")
}

model VerificationRequest {
  id         Int      @id @default(autoincrement())
  identifier String
  token      String   @unique
  expires    DateTime
  createdAt  DateTime @default(now()) @map(name: "created_at")
  updatedAt  DateTime @default(now()) @map(name: "updated_at")

  @@map(name: "verification_requests")
}

model Subreddit {
  id          Int      @id @default(autoincrement())
  name        String   @unique
  displayName String
  infoBoxText String
  createdAt   DateTime @default(now())
  posts       Post[]
  joinedUsers User[]
}

model Post {
  id          Int       @id @default(autoincrement())
  title       String
  body        String
  subreddit   Subreddit @relation(fields: [subredditId], references: [id])
  comments    Comment[]
  user        User      @relation(fields: [userId], references: [id])
  votes       Vote[]
  subredditId Int
  userId      Int
  createdAt   DateTime  @default(now())
  updatedAt   DateTime  @updatedAt
}

model Comment {
  id     Int    @id @default(autoincrement())
  body   String
  user   User   @relation(fields: [userId], references: [id])
  votes  Vote[]
  post   Post   @relation(fields: [postId], references: [id])
  userId Int
  postId Int
}

model Vote {
  id        Int      @id @default(autoincrement())
  voteType  VoteType
  user      User     @relation(fields: [userId], references: [id])
  post      Post?    @relation(fields: [postId], references: [id])
  comment   Comment? @relation(fields: [commentId], references: [id])
  userId    Int
  postId    Int?
  commentId Int?
}

enum VoteType {
  UPVOTE
  DOWNVOTE
}
```

Also anybody knows how to write SQL Query with arguments ($1 for example) in pgAdmin? 😃 

Thx SelfDevTV

**Comment by ryands17** [comment, marked as answer]
Hey @SelfDevTV 👋 
The table names above are due to a convention Prisma follows for implicit many to many relations that you can read about [here](https://www.prisma.io/docs/concepts/components/prisma-schema/relations#conventions-for-relation-tables-in-implicit-m-n-relations).

Also for PgAdmin you would need to use [Prepared Statements](https://www.postgresql.org/docs/current/sql-prepare.html).

**Reply by SelfDevTV** [reply]
Hiho @ryands17 🤝 

Thank you for providing this link! That helps a lot. I wasn't able to find it. Also thank you for the Prepared Statements link. 

As always you helped a ton! Maybe you could look into my performance issue real quick #4884 . I have provided tons of information. The last update I posted I created an API Route that returns the miliseconds of the prisma execution time. Maybe give it the performance tag so it gets redirected to the correct "team".
If you don't have time for it no big deal.

As always keep up the good work!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
